### PR TITLE
boards: Update SAM0-based documentation

### PR DESCRIPTION
**Pull request by dylad**
### Contribution description
This PR updates some sam0-based boards documentation as some of them were quite outdated.

If you want me to add specific information, feel free to speak up.

### Testing procedure
Look at the rendered documentation by CI.

### Issues/PRs references
None.

### CODE REVIEW

File: boards/samd10-xmini/doc.txt [hunk 1]
```diff
@@ -50,7 +50,13 @@ The samd10-xmini is available from various hardware vendors for ~8€ (as of
 Connect the device to your Micro-USB cable using the port labeled as *DEBUG
 USB*.
 
-The standard method for flashing RIOT to the samd10-xmini is using EDBG.
+The standard method for flashing RIOT to the samd21-xpro is using [edbg](https://github.com/ataradov/edbg).
```
Comment (benpicco) [inline boards/samd10-xmini/doc.txt:53]:
```suggestion
The standard method for flashing RIOT to the samd10-xmini is using [edbg](https://github.com/ataradov/edbg).
```

File: boards/samd20-xpro/doc.txt [hunk 2]
```diff
@@ -47,11 +47,13 @@ memory.
 Connect the device to your Micro-USB cable using the port labeled as *DEBUG
 USB*.
 
-The standard method for flashing RIOT to the samd20-xpro is using OpenOCD.
+The standard method for flashing RIOT to the samd21-xpro is using [edbg](https://github.com/ataradov/edbg).
```
Comment (benpicco) [inline boards/samd20-xpro/doc.txt:50]:
```suggestion
The standard method for flashing RIOT to the samd20-xpro is using [edbg](https://github.com/ataradov/edbg).
```

File: boards/samd20-xpro/doc.txt [hunk 3]
```diff
@@ -47,11 +47,13 @@ memory.
...
+by calling: `make BOARD=samd21-xpro -C tests/leds flash`
 
-Refer to https://github.com/RIOT-OS/RIOT/wiki/OpenOCD for general
-instructions on building OpenOCD and make sure "cmsis-dap" and "hidapi-libusb"
-are enabled.
+Note that on Linux, you will need libudev-dev package to be installed.
+
+Users can also use openOCD to flash and/or debug the board using:
+`PROGRAMMER=openocd make BOARD=saml21-xpro -C tests/leds flash`
```
Comment (benpicco) [inline boards/samd20-xpro/doc.txt:56]:
```suggestion
`PROGRAMMER=openocd make BOARD=samd20-xpro -C tests/leds flash`
```

File: boards/samd20-xpro/doc.txt [hunk 4]
```diff
@@ -77,7 +80,7 @@ the SAMD20: http://watr.li/samr21-dev-setup-ubuntu.html
 ## Supported Toolchains
 
 For using the samd20-xpro board we strongly recommend the usage of the
-[GNU Tools for ARM Embedded Processors](https://launchpad.net/gcc-arm-embedded)
+[GNU Tools for ARM Embedded Processors](https://developer.arm.com/Tools%20and%20Software/GNU%20Toolchain)
 toolchain.
```
Comment (benpicco) [inline boards/samd20-xpro/doc.txt:84]: I'd drop that, any `arm-none-eabi-gcc` is fine - I've been using the one from the Ubuntu repo for years.

File: boards/samd21-xpro/doc.txt [hunk 5]
```diff
@@ -50,31 +50,30 @@ The samd21-xpro is available from various hardware vendors for ~30USD (as of
...
+The standard method for flashing RIOT to the samd21-xpro is using [edbg](https://github.com/ataradov/edbg).
+by calling: `make BOARD=samd21-xpro -C tests/leds flash`
+
+Note that on Linux, you will need libudev-dev package to be installed.
 
-Refer to https://github.com/RIOT-OS/RIOT/wiki/OpenOCD for general
-instructions on building OpenOCD and make sure "cmsis-dap" and "hidapi-libusb"
-are enabled.
+Users can also use openOCD to flash and/or debug the board using:
+`PROGRAMMER=openocd make BOARD=saml21-xpro -C tests/leds flash`
```
Comment (benpicco) [inline boards/samd21-xpro/doc.txt:76]:
```suggestion
`PROGRAMMER=openocd make BOARD=samd21-xpro -C tests/leds flash`
```

File: boards/same54-xpro/doc.txt [hunk 6]
```diff
@@ -73,7 +73,14 @@ The same54-xpro is available from various hardware vendors for ~70USD (as of
 Connect the device to your Micro-USB cable using the port labeled as *DEBUG
 USB*.
 
-The standard method for flashing RIOT to the same54-xpro is using EDBG.
+The standard method for flashing RIOT to the samd21-xpro is using [edbg](https://github.com/ataradov/edbg).
```
Comment (benpicco) [inline boards/same54-xpro/doc.txt:76]:
```suggestion
The standard method for flashing RIOT to the same54-xpro is using [edbg](https://github.com/ataradov/edbg).
```

**Comment by dylad** [comment]
@benpicco I've addressed your comments.

**Review by kfessel — COMMENTED** [review]
please squash your fixups 

I am also not sure about the tool-chain recommendation (sometimes there is one, sometimes not, which one seems a little random)
but on the other hand this improves and updates the exiting documentation and that is what this PR aimed for

**Comment by dylad** [comment]
> I am also not sure about the tool-chain recommendation (sometimes there is one, sometimes not, which one seems a little random)
Maybe I miss some toolchain removals, I'll fix that to remove all references then.

**Comment by dylad** [comment]
Squashed. I also remove all 'preferred toolchain' reference from the boards documentation.

**Review by kfessel — APPROVED** [review]

**Pull request merged**